### ISSUE TITLE
Unexpected behavior of 'ex' option when argument contains parenthesis

### ISSUE THREAD

**Issue by LLCampos** [etorreborre/specs2#956]
If the argument for `ex` option contains an opening parenthesis (`(`), it doesn't match any test.

Version 4.12.0.

# How to reproduce:

```scala
import org.specs2.mutable.Specification

class MySpec extends Specification {
  
  "Top level" should {
    "first test (1)" in {
      ok
    }
  }
}
```

**No parenthesis :heavy_check_mark:**

```sbt> testOnly *MySpec* -- ex "first test"```

```
[info] MySpec
[info] Top level should
[info]   + first test (1)
[info] Total for specification MySpec
[info] Finished in 202 ms
[info] 1 example, 0 failure, 0 error
[info] Passed: Total 1, Failed 0, Errors 0, Passed 1
[success] Total time: 2 s, completed 24/06/2021, 14:53:06
```

**With parenthesis :x:**

```sbt> testOnly *MySpec* -- ex "first test (1)"```

```
[info] MySpec
[info] Top level should
[info] Total for specification MySpec
[info] Finished in 168 ms
[info] 0 example, 0 failure, 0 error
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[success] Total time: 1 s, completed 24/06/2021, 14:54:51
```

**With only the opening parenthesis :x:**

```sbt> testOnly *MySpec* -- ex "first test ("```

```
[info] MySpec
[info] Top level should
[info] Total for specification MySpec
[info] Finished in 165 ms
[info] 0 example, 0 failure, 0 error
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[success] Total time: 2 s, completed 24/06/2021, 14:55:53
```

I would expect all the commands above to return the same thing, but the two last ones don't run the test.

**Comment by etorreborre** [comment]
That's because specs2 does something which I thought would be more useful. The first thing to know is that the value passed to `ex` is interpreted as a regular expression, `.*<ex value>.*` so that you don't have to type the full name all the time. For example `ex 1` or `ex "first test"` would work.

But if that pattern is not a valid regular expression (i.e. it can't be compiled with `java.util.regex.Pattern.compile`) then we quote the whole expression using `Pattern.quote`, trying to interpret the string literally. But in that case this brings `"\Q.*first test(\E"` which cannot match. We need to quote as `".*\Qfirst test(\E."`. This is the fix I am going to do.

In the meantime the workaround is to pass `ex "\\\\Qfirst test (\\\\E"` to quote the string yourself (yes we need 4 `\`)

**Comment by etorreborre** [comment]
This is fixed now in `4.12.2`
